app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -159,6 +177,23 @@ fun HomeScreen(
               start.linkTo(parent.start)
               end.linkTo(parent.end)
             })
+
+    var viewToggleIcon =
+        if (viewToggleBrowseIndex == 0) Icons.Default.Place else Icons.AutoMirrored.Filled.List
+    FloatingActionButton(
+        onClick = {
+          viewToggleBrowseIndex = if (viewToggleBrowseIndex == 0) 1 else 0
+          viewToggleIcon =
```

Comment:
nit: The var "viewToggleIcon" may not be necessary. Compose's reactivity means that the UI will redraw whenever state changes, so you can define the Icon directly based on viewToggleBrowseIndex when the icon is being drawn, not when the button is clicked. Like this: 
 Icon(
        imageVector = if (viewToggleBrowseIndex == 0) Icons.Default.Place else Icons.AutoMirrored.Filled.List,
        contentDescription = null
    )

Reply:
True, I decided to write it this way mostly to be a little more verbose in the code. Also, I didn't want to have wait until the view was recomposed for the icon to change.